lib/cortex/file_watcher.ex

```diff
@@ -26,14 +31,21 @@ defmodule Cortex.FileWatcher do
 
     FileSystem.subscribe(watcher_pid)
 
-    {:ok, %{watcher_pid: watcher_pid}}
+    {:ok, %State{watcher_pid: watcher_pid, throttle_timer: nil, file_events: %{}}}
   end
 
   def handle_info(
         {:file_event, watcher_pid, {path, _events}},
         %{watcher_pid: watcher_pid} = state
       ) do
-    GenServer.cast(Controller, {:file_changed, file_type(path), path})
+    %State{file_events: file_events, throttle_timer: throttle_timer} = state
+
+    unless throttle_timer do
```

Comment:
hmm, maybe I'm missing something, but does this ever get set to anything other than `nil`?

Reply:
Nope, that definitely looks like an oversight. Will fix it up in a bit :+1:

Reply:
Okay, code is updated to actually set and clear the timer.